src/from_parentdir.py

```diff
@@ -1,11 +1,18 @@
 
 def versions_from_parentdir(parentdir_prefix, root, verbose=False):
     # Source tarballs conventionally unpack into a directory that includes
-    # both the project name and a version string.
-    dirname = os.path.basename(root)
-    if not dirname.startswith(parentdir_prefix):
-        if verbose:
-            print("guessing rootdir is '%s', but '%s' doesn't start with prefix '%s'" %
-                  (root, dirname, parentdir_prefix))
-        return None
-    return {"version": dirname[len(parentdir_prefix):], "full": ""}
+    # both the project name and a version string. We will also support searching
+    # up two directory levels for an appropriately named parent directory
+    rootdirs = []
+
+    for i in range(3):
```

Comment:
Is there a special reason for the limitation to 3 parent directories? What if I have a deeper nested structure?